lightly/data/_video.py

```diff
@@ -47,6 +48,117 @@ def _video_loader(path, timestamp, pts_unit='sec'):
+        transform:
+            Function that takes a PIL image and returns transformed version
+        target_transform:
+            As transform but for targets
+        is_valid_file:
+            Used to check corrupt files
+
+    Examples:
+        >>> from torchvision import io
+        >>>
+        >>> # get timestamps
+        >>> ts, fps = io.read_video_timestamps('myvideo.mp4', pts_unit = 'sec')
+        >>>
+        >>> # create a VideoLoader
+        >>> video_loader = VideoLoader('myvideo.mp4', ts)
+        >>>
+        >>> # get frame at specific timestamp
+        >>> frame = video_loader.read_frame(ts[21])
+        >>>
+        >>> # get next frame
+        >>> frame = video_loader.read_frame()
+    """
+    def __init__(self, path: str, timestamps: List[float], backend: str = 'video_reader'):
+        self.path = path
+        self.timestamps = timestamps
+        self.current_timestamp_idx = 0
+        self.last_timestamp_idx = 0
+        self.pts_unit='sec'
+        self.backend = backend
+
+        has_video_reader = io._HAS_VIDEO_OPT and hasattr(io, 'VideoReader')
+
+        if has_video_reader and self.backend == 'video_reader':
+            self.reader = io.VideoReader(path = self.path)
+        else:
+            self.reader = None
+    
+    def read_frame(self, timestamp = None):
+        """Reads the next frame or from timestamp.
+
+        If no timestamp is provided this method just returns the next frame from
+        the video. This is significantly (up to 10x) faster if the `video_loader` 
+        backend is available. If a timestamp is provided we first have to seek
+        to the right position and then load the frame.
+        
+        Args:
+            timestamp: Specific timestamp of frame in seconds or None (default: None)
+
+        Returns:
+            A PIL Image
+
+        """
+        if timestamp != None:
+            self.current_timestamp_idx = self.timestamps.index(timestamp)
+        else:
+            if self.current_timestamp_idx < len(self.timestamps):
+                self.current_timestamp_idx += 1
+
+        if self.reader:
+            if timestamp != None:
+                if self.timestamps.index(timestamp) != self.last_timestamp_idx + 1:
+                    self.reader.seek(timestamp)
+
+            # make sure we have the tensor in correct shape (we want H x W x C)
+            frame = next(self.reader)['data'].permute(1,2,0)
+            self.last_timestamp_idx = self.current_timestamp_idx
+
+        else: # fallback on pyav
+            if timestamp == None:
+                timestamp = self.timestamps[self.current_timestamp_idx]
+            frame, _, _ = io.read_video(self.path,
+                                        start_pts=timestamp,
+                                        end_pts=timestamp,
+                                        pts_unit=self.pts_unit)    
+            self.last_timestamp_idx = self.timestamps.index(timestamp)    
+        
```

Comment:
Nice idea! I have some feedback:
- Be consistent in the way you check for `None` (I would always go with `if x is None` or `if x is not None`).
- A few comments would make reading this part a lot easier.